metadata-ingestion/src/datahub/ingestion/source/affirm/artifact.py

```diff
@@ -36,27 +38,34 @@ def __post_init__(self):
             self.privacy_entrypoint = ''
         if self.processing_purposes is None:
             self.processing_purposes = []
+        if self.schema_name is None:
+            self.schema_name = ''
 
 
-class AffirmArtifactSourceConfig(ConfigModel):
+class AffirmArtifactSourceConfig(PlatformSourceConfigBase):
     ''' TODO support git repo to automate the whole process: 
     git clone, locate artifact and ingest
     '''
     directory: str
-    platform: str
     env: str
 
 
 def iterate_artifact(directory: str) -> Iterable[AffirmArtifact]:
     def fix_description(description: str):
         return ' '.join(description.split())
-    for r, _, filenames in os.walk(directory):
+    def get_schema(dir: str):
+        relative_dir = dir.replace(os.path.abspath(directory), '').strip()
+        relative_dir = relative_dir if not relative_dir.startswith(os.sep) else relative_dir[len(os.sep):]
+        relative_dir = relative_dir if not relative_dir.endswith(os.sep) else relative_dir[len(os.sep):]
+        return '.'.join(relative_dir.split(os.sep))
```

Comment:
Can we expect a git directory and get the git root to get the relative path instead?